### PR TITLE
Fix double directory fetch

### PR DESCRIPTION
**Pull request by johnjeng**
Prevent double directory fetching inside certain actions

### CODE REVIEW

File: src/smc-webapp/project_actions.ts [hunk 1]
```diff
@@ -1130,7 +1133,7 @@ export class ProjectActions extends Actions<ProjectStoreState> {
         if (store.get("active_project_tab") === "files") {
           this.set_url_to_path(path);
         } else {
-          this.set_active_tab("files");
+          this.set_active_tab("files", {update_file_listing: false});
```
Comment (williamstein) [inline src/smc-webapp/project_actions.ts]: Why do you NOT want to update the file listing here?

File: src/smc-webapp/project_actions.ts [hunk 2]
```diff
@@ -352,7 +352,7 @@ export class ProjectActions extends Actions<ProjectStoreState> {
   //            or a file_redux_name
   // Pushes to browser history
   // Updates the URL
-  set_active_tab(key: string): void {
+  set_active_tab(key: string, opts: {update_file_listing: boolean} = {update_file_listing: true}): void {
```
Comment (williamstein) [inline src/smc-webapp/project_actions.ts]: 100% of other users of opts in CoCalc have opts as the *only option* to the function.  Don't missing positional args and opts.  Use one or the other.    This is also a weird use of opts, where this is only one option?

**Comment by williamstein** [comment]
It doesn't work:

screenshot 2018-08-31 at 7 48 17 am

File: src/smc-webapp/project_actions.ts (hunk 1 above)
Comment (johnjeng) [reply src/smc-webapp/project_actions.ts]: Because `set_current_path` also updates the file listing.

**Pull request merged**